### PR TITLE
adding feedback modal for students

### PR DESCRIPTION
**Pull request by eric-ls**
screen shot 2017-02-19 at 7 05 27 pm
screen shot 2017-02-19 at 7 05 45 pm

### CODE REVIEW

**Comment by eric-ls** [comment]
@echang9 this mostly works! need some feedback on my changes to `renderButton()` in the lessonCard. I also need to manually create a new lesson and test

File: app/assets/javascripts/components/dashboard/feedback_modal.es6.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+class FeedbackModal extends React.Component {
+
+  constructor(props) {
+    console.log(props)
```
Comment (echang9) [inline app/assets/javascripts/components/dashboard/feedback_modal.es6.jsx]: probs dont need this :p

File: app/assets/javascripts/components/dashboard/feedback_modal.es6.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      <div>
+        <Modal.Body>
+          <p>Thanks for using Forte! Please leave some feedback
+          about your most recent lesson on <strong>{lessonTime}</strong>.</p>
+
+          <p>Your feedback is extremely valuable and helps us deliver a better experience!</p>
+
+          <FormGroup className="marginTop-md">
+            <ControlLabel>Comments</ControlLabel>
+            <p className="form-input-description">Your teacher won't see what you write here.</p>
```
Comment (echang9) [inline app/assets/javascripts/components/dashboard/feedback_modal.es6.jsx]: Perhaps a softer description? "Only Forte Admins will see what you write here!"

File: app/assets/javascripts/components/student/student_lessons_page.es6.jsx [hunk 3]
```diff
@@ -44,6 +48,38 @@ class StudentLessonsPage extends React.Component {
     );
   }
 
+  openFeedbackModal() {
```
Comment (echang9) [inline app/assets/javascripts/components/student/student_lessons_page.es6.jsx]: this is so dope 👍. localStorage yeee

File: app/assets/javascripts/components/student/student_lessons_page.es6.jsx [hunk 4]
```diff
@@ -44,6 +48,38 @@ class StudentLessonsPage extends React.Component {
...
+    // Check cache
+    const shown = JSON.parse(localStorage.getItem("shownModal"));
+    const storedRecent = JSON.parse(localStorage.getItem("recentLesson"));
+    const mostRecent = this.state.recentLessons[0];
+
+    if (!mostRecent) { return; }
+
+    if (!storedRecent == mostRecent.id) {
```
Comment (echang9) [inline app/assets/javascripts/components/student/student_lessons_page.es6.jsx]: shouldn't we do a check to see ```if (!storedRecent == mostRecent.id && !shown)```

File: app/assets/javascripts/components/student/student_lessons_page.es6.jsx [hunk 5]
```diff
@@ -44,6 +48,38 @@ class StudentLessonsPage extends React.Component {
     );
   }
 
+  openFeedbackModal() {
+    // Check cache
+    const shown = JSON.parse(localStorage.getItem("shownModal"));
+    const storedRecent = JSON.parse(localStorage.getItem("recentLesson"));
+    const mostRecent = this.state.recentLessons[0];
+
+    if (!mostRecent) { return; }
```
Comment (echang9) [inline app/assets/javascripts/components/student/student_lessons_page.es6.jsx]: Nice!!!

File: app/assets/javascripts/components/dashboard/lesson_card.es6.jsx [hunk 6]
```diff
@@ -86,41 +95,73 @@ class LessonCard extends React.Component {
...
+        <FeedbackModal
+          handleClose={() => this.closeFeedbackModal()}
+          lesson={this.props.lesson}
+        />
+      )
+    }
+  }
+
   renderButtons() {
+    let cancelBtn, rescheduleBtn, feedbackBtn, payBtn;
```
Comment (echang9) [inline app/assets/javascripts/components/dashboard/lesson_card.es6.jsx]: this is dope too 💯

**Pull request merged**